### PR TITLE
Small fixes: breath-update, workaround for clang-12 errors & adding includes for gcc-11

### PR DESCRIPTION
**Pull request by fabratu**
This PR introduces the following fixes:

- Introducing workarounds for clang-12 errors in `GraphGTest`, `GraphBuilderAutoCompleteGTest`and `GraphBuilderDirectSwapGTest` by rewriting the loops and removing `critical`-sections.
- Introducing workarounds for clang-12 error in `LocalSimilarityScore` by exchanging function call of `balancedParallelForNodes` by it's logic. 
- The pinning for breathe is removed, since the new release doesn't break anymore. 
- Slight changes to documentation code (see suggestions in https://github.com/michaeljones/breathe/releases/tag/v4.29.2). 
- Add explicit includes for `<limits>` due to changes in gcc-11 (errors already visible on CI for most recent Ubuntu-runners) (see #755)

Background to clang-12 errors:

The most recent version (12.0.0 & 12.0.1) crashes on IR generation for four occasions: [GraphGTest.cpp (line 1346)](https://github.com/networkit/networkit/blob/master/networkit/cpp/graph/test/GraphGTest.cpp#L1346), [GraphBuilderAutoCompleteGTest.cpp (line 486)](https://github.com/networkit/networkit/blob/master/networkit/cpp/graph/test/GraphBuilderAutoCompleteGTest.cpp#486), [GraphBuilderDirectSwapGTest.cpp (line 531)](https://github.com/networkit/networkit/blob/master/networkit/cpp/graph/test/GraphBuilderDirectSwapGTest.cpp#L531), [LocalSimilarityScore.cpp (line 51)](https://github.com/networkit/networkit/blob/master/networkit/cpp/sparsification/LocalSimilarityScore.cpp#L51). In all cases this happens during `parallelForNodes`, after `#pragma omp critical` is set. All other usages of `critical` don't involve `parrallelForNodes` (and don't cause crashes). If `critical` is removed (which we don't want of course), building the code completes without error. The errors are very likely bugs, the code itself seems ok.

### CODE REVIEW

**Comment by angriman** [comment]
Not being able to use a critical section within a `parallelForNodes` loop sounds restrictive. Do you know why those crashes happen? Do we have any other alternative solution?

File: networkit/cpp/graph/test/GraphBuilderAutoCompleteGTest.cpp [hunk 1]
```diff
@@ -483,9 +483,9 @@ TEST_P(GraphBuilderAutoCompleteGTest, testForNodes) {
 
 TEST_P(GraphBuilderAutoCompleteGTest, testParallelForNodes) {
     std::vector<node> visited;
+    visited.resize(bHouse.upperNodeIdBound());
```
Comment (angriman) [inline networkit/cpp/graph/test/GraphBuilderAutoCompleteGTest.cpp:486]: You can drop the `resize` by passing the size to the constructor of the vector.

File: networkit/cpp/graph/test/GraphBuilderDirectSwapGTest.cpp [hunk 2]
```diff
@@ -528,9 +528,9 @@ TEST_P(GraphBuilderDirectSwapGTest, testForNodes) {
 
 TEST_P(GraphBuilderDirectSwapGTest, testParallelForNodes) {
     std::vector<node> visited;
+    visited.resize(bHouse.upperNodeIdBound());
```
Comment (angriman) [inline networkit/cpp/graph/test/GraphBuilderDirectSwapGTest.cpp:531]: Same as above.

File: networkit/cpp/graph/test/GraphGTest.cpp [hunk 3]
```diff
@@ -1343,9 +1343,9 @@ TEST_P(GraphGTest, testForNodes) {
 
 TEST_P(GraphGTest, testParallelForNodes) {
     std::vector<node> visited;
+    visited.resize(Ghouse.upperNodeIdBound());
```
Comment (angriman) [inline networkit/cpp/graph/test/GraphGTest.cpp:1346]: Also here.

**Comment by fabratu** [comment]
> Do you know why those crashes happen? Do we have any other alternative solution?

I think all occasions are bugs, since clang crashes on IR-generation. The only alternative solution I came up with, is to have workarounds. Thats means either rewriting the critical-section (while still `parallelForNodes`) or exchange the call to `parallelForNodes` with it's logic. Both approaches are used in the PR. As a result, clang-12 compiles correctly.

**Comment by avdgrinten** [comment]
Are the bugs reported to Clang's upstream?

**Comment by fabratu** [comment]
> Are the bugs reported to Clang's upstream?

Not yet, currently working on it.

**Review by angriman — APPROVED** [review]
I think we can merge this, I don't like the workaround for critical sections in `parallelForNodes`, but on the other hand it affects very few parts of the code base and I don't see alternative solutions.

**Comment by fabratu** [comment]
Will merge this early (small fixes, needed for correct CI-tests)

**Pull request merged**